i hope the thingy will not complain about this 3 vars , i was thining to cover those in a separte MR 

```
[vmindru@vmutil ansible-devel]$ flake8 lib/ansible/modules/packaging/os/yum.py 
lib/ansible/modules/packaging/os/yum.py:829:13: F841 local variable 'is_group' is assigned to but never used
lib/ansible/modules/packaging/os/yum.py:1014:17: F841 local variable 'msg' is assigned to but never used
lib/ansible/modules/packaging/os/yum.py:1128:29: F841 local variable 'a' is assigned to but never used
[vmindru@vmutil ansible-devel]$ 
````

--- a/lib/ansible/modules/packaging/os/yum.py
+++ b/lib/ansible/modules/packaging/os/yum.py
@@ -1103,25 +1103,18 @@ def ensure(module, state, pkgs, conf_file, enablerepo, disablerepo,
             module.run_command(yum_basecmd + ['clean', 'expire-cache'])
 
         my = yum_base(conf_file, installroot)
-        try:
-            if disablerepo:
+        if disablerepo:
+            try:
                 my.repos.disableRepo(disablerepo)
-            current_repos = my.repos.repos.keys()
-            if enablerepo:
-                try:
-                    my.repos.enableRepo(enablerepo)
-                    new_repos = my.repos.repos.keys()
-                    for i in new_repos:
-                        if not i in current_repos:
-                            rid = my.repos.getRepo(i)
-                            a = rid.repoXML.repoid  ##  if no one complains remove this on next MR
-                    current_repos = new_repos
-                except yum.Errors.YumBaseError:
-                    e = get_exception()
-                    module.fail_json(msg="Error setting/accessing repos: %s" % (e))
-        except yum.Errors.YumBaseError:
-            e = get_exception()
-            module.fail_json(msg="Error accessing repos: %s" % e)
+            except yum.Errors.YumBaseError:
+                yum_error = get_exception()
+                module.fail_json(msg="Error setting/accessing repos: %s" % (yum_error))
+        if enablerepo:
+            try:
+                my.repos.enableRepo(enablerepo)
+            except yum.Errors.YumBaseError:
+                yum_error = get_exception()
+                module.fail_json(msg="Error setting/accessing repos: %s" % (yum_error))
     if state in ['installed', 'present']:
         if disable_gpg_check:
             yum_basecmd.append('--nogpgcheck')

--- a/lib/ansible/modules/packaging/os/yum.py
+++ b/lib/ansible/modules/packaging/os/yum.py
@@ -643,18 +643,8 @@ def list_stuff(module, repoquerybin, conf_file, stuff, installroot='/'):
     elif stuff == 'repos':
         return [ dict(repoid=name, state='enabled') for name in sorted(repolist(module, repoq)) if name.strip() ]
     else:
-        return [ pkg_to_dict(p) for p in sorted(is_installed(module,
-                                                             repoq,
-                                                             stuff,
-                                                             conf_file,
-                                                             qf=is_installed_qf,
-                                                             installroot=installroot) +
-                                                is_available(module,
-                                                             repoq,
-                                                             stuff,
-                                                             conf_file,
-                                                             qf=qf,
-                                                             installroot=installroot)) if p.strip() ]
+        return [ pkg_to_dict(p) for p in sorted(is_installed(module,repoq, stuff, conf_file, qf=is_installed_qf, installroot=installroot)+
+                                                is_available(module, repoq, stuff, conf_file, qf=qf, installroot=installroot)) if p.strip()]
 
 def install(module, items, repoq, yum_basecmd, conf_file, en_repos, dis_repos, installroot='/'):
 
@@ -1011,7 +1001,6 @@ def latest(module, items, repoq, yum_basecmd, conf_file, en_repos, dis_repos, in
         for w in will_update:
             if w.startswith('@'):
                 to_update.append((w, None))
-                res['msg'] += '%s will be updated' % w
             elif w not in updates:
                 other_pkg = will_update_from_other_package[w]
                 to_update.append((w, 'because of (at least) %s-%s.%s from %s' % (other_pkg,
@@ -1125,7 +1114,7 @@ def ensure(module, state, pkgs, conf_file, enablerepo, disablerepo,
                     for i in new_repos:
                         if not i in current_repos:
                             rid = my.repos.getRepo(i)
-                            a = rid.repoXML.repoid  ##  if no one complains remove this on next MR 
+                            a = rid.repoXML.repoid  ##  if no one complains remove this on next MR
                     current_repos = new_repos
                 except yum.Errors.YumBaseError:
                     e = get_exception()
@@ -1151,6 +1140,18 @@ def ensure(module, state, pkgs, conf_file, enablerepo, disablerepo,
 
 
 def main():
+
+    # state=installed name=pkgspec
+    # state=removed name=pkgspec
+    # state=latest name=pkgspec
+    #
+    # informational commands:
+    #   list=installed
+    #   list=updates
+    #   list=available
+    #   list=repos
+    #   list=pkgspec
+
 
     module = AnsibleModule(
         argument_spec = dict(

--- a/lib/ansible/modules/packaging/os/yum.py
+++ b/lib/ansible/modules/packaging/os/yum.py
@@ -1011,7 +1011,7 @@ def latest(module, items, repoq, yum_basecmd, conf_file, en_repos, dis_repos, in
         for w in will_update:
             if w.startswith('@'):
                 to_update.append((w, None))
-                msg = '%s will be updated' % w
+                res['msg'] += '%s will be updated' % w
             elif w not in updates:
                 other_pkg = will_update_from_other_package[w]
                 to_update.append((w, 'because of (at least) %s-%s.%s from %s' % (other_pkg,
@@ -1086,7 +1086,7 @@ def ensure(module, state, pkgs, conf_file, enablerepo, disablerepo,
     en_repos = []
 
     if skip_broken:
-        yum_basecmd.extend('--skip-broken')
+        yum_basecmd.extend(['--skip-broken'])
 
     if disablerepo:
         dis_repos = disablerepo.split(',')
@@ -1125,7 +1125,7 @@ def ensure(module, state, pkgs, conf_file, enablerepo, disablerepo,
                     for i in new_repos:
                         if not i in current_repos:
                             rid = my.repos.getRepo(i)
-                            a = rid.repoXML.repoid
+                            a = rid.repoXML.repoid  ##  if no one complains remove this on next MR 
                     current_repos = new_repos
                 except yum.Errors.YumBaseError:
                     e = get_exception()
@@ -1147,7 +1147,6 @@ def ensure(module, state, pkgs, conf_file, enablerepo, disablerepo,
         # should be caught by AnsibleModule argument_spec
         module.fail_json(msg="we should never get here unless this all"
                 " failed", changed=False, results='', errors='unexpected state')
-
     return res
 
 

